File: backend/src/main/java/com/gxs/bank/controller/BeneficiaryController.java
package com.gxs.bank.controller;

import com.gxs.bank.dto.response.ApiResponse;
import com.gxs.bank.model.Beneficiary;
import com.gxs.bank.model.User;
import com.gxs.bank.service.BeneficiaryService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.UUID;

@RestController
@RequestMapping("/api/beneficiaries")
@RequiredArgsConstructor
public class BeneficiaryController {
    private final BeneficiaryService beneficiaryService;

    @PostMapping
    public ResponseEntity<ApiResponse<Beneficiary>> addBeneficiary(@AuthenticationPrincipal User user, @RequestBody Beneficiary request) {
        Beneficiary ben = beneficiaryService.addBeneficiary(user.getId(), request);
        return ResponseEntity.ok(ApiResponse.ok("Beneficiary added successfully", ben));
    }

    @GetMapping
    public ResponseEntity<ApiResponse<List<Beneficiary>>> getBeneficiaries(@AuthenticationPrincipal User user) {
        return ResponseEntity.ok(ApiResponse.ok(beneficiaryService.getUserBeneficiaries(user.getId())));
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<ApiResponse<String>> deleteBeneficiary(@AuthenticationPrincipal User user, @PathVariable UUID id) {
        beneficiaryService.deleteBeneficiary(id, user.getId());
        return ResponseEntity.ok(ApiResponse.ok("Beneficiary deleted successfully"));
    }
}

File: backend-python/com/gxs/bank/controller/BeneficiaryController.py
from fastapi import APIRouter, Depends

from com.gxs.bank.config.database import get_db
from com.gxs.bank.config.serialization import serialize
from com.gxs.bank.dto.response.ApiResponse import ok
from com.gxs.bank.model.Beneficiary import Beneficiary
from com.gxs.bank.security.JwtAuthenticationFilter import get_current_user
from com.gxs.bank.service.BeneficiaryService import BeneficiaryService


router = APIRouter(prefix="/api/beneficiaries", tags=["Beneficiary"])


@router.post("")
def addBeneficiary(request: dict, current_user=Depends(get_current_user), db=Depends(get_db)):
    ben = Beneficiary(**request)
    saved = BeneficiaryService(db).addBeneficiary(current_user.id, ben)
    return ok(serialize(saved), "Beneficiary added successfully")


@router.get("")
def getBeneficiaries(current_user=Depends(get_current_user), db=Depends(get_db)):
    beneficiaries = BeneficiaryService(db).getUserBeneficiaries(current_user.id)
    return ok(serialize(beneficiaries))


@router.delete("/{beneficiary_id}")
def deleteBeneficiary(beneficiary_id: str, current_user=Depends(get_current_user), db=Depends(get_db)):
    BeneficiaryService(db).deleteBeneficiary(beneficiary_id, current_user.id)
    return ok("Beneficiary deleted successfully")
